File: src/content/3.4/code/scala/snippet01.scala
def fmap[A, B]: (A => B) => List[A] => List[B] =
  f => {
    case Nil => Nil
    case x :: xs => f(x) :: fmap[A, B](f)(xs)
  }

def sum: List[Double] => Double = _.sum

import math._

def vlen: List[Double] => Double =
  sqrt _ compose sum compose fmap(pow(_, 2))

File: src/content/3.4/code/ocaml/snippet01.ml
let ( % ) = Fun.compose

(* There's no sum function in the List module *)
let sum = List.fold_left ( +. ) 0.

(* The vlen function, like the Haskell version, using
   List.map as the list functor fmap *)
let vlen = sqrt % sum % List.map (Fun.flip ( ** ) 2.)
